tests/python_api/jax_classification_reader.py

```diff
@@ -0,0 +1,109 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import random
+from amd.rocal.plugin.jax import ROCALJaxIterator
+from amd.rocal.pipeline import Pipeline
+import amd.rocal.fn as fn
+import amd.rocal.types as types
+import numpy as np
+import sys
+import os
+
+import jax
+from jax.sharding import Mesh, PartitionSpec, NamedSharding
+
+
+def draw_patches(img, idx, layout="nchw", dtype="fp32"):
+    # image is expected as a tensor
+    import cv2
+    image = np.asarray(img)
+    if layout == "nchw":
+        image = image.transpose([1, 2, 0])
+    if dtype == "fp16":
+        image = image.astype("uint8")
+    image = cv2.cvtColor(image, cv2.COLOR_RGB2BGR)
+    cv2.imwrite("output_folder/jax_reader/" + str(idx) +
+                "_" + "train" + ".png", image * 255)
+
+
+def main():
+    if len(sys.argv) < 3:
+        print("Please pass image_folder cpu/gpu batch_size")
+        exit(0)
+    try:
+        path = "output_folder/jax_reader/"
+        isExist = os.path.exists(path)
+        if not isExist:
+            os.makedirs(path)
+    except OSError as error:
+        print(error)
+    data_path = sys.argv[1]
+    rocal_cpu = True if sys.argv[2] == "cpu" else False
+    batch_size = int(sys.argv[3])
+    num_threads = 8
+    random_seed = random.SystemRandom().randint(0, 2**32 - 1)
+
+    mesh = Mesh(jax.devices(), axis_names=("batch"))
+    sharding = NamedSharding(mesh, PartitionSpec("batch"))
+    pipelines = []
+    for id, device in enumerate(jax.devices()):
+        image_classification_train_pipeline = Pipeline(batch_size=batch_size, num_threads=num_threads, device_id=id,
+                                                       seed=random_seed, rocal_cpu=rocal_cpu, tensor_layout=types.NHWC, tensor_dtype=types.FLOAT16)
+
+        with image_classification_train_pipeline:
+            jpegs, labels = fn.readers.file(file_root=data_path)
+            decode = fn.decoders.image_slice(jpegs, output_type=types.RGB,
+                                             file_root=data_path, shard_id=id, num_shards=jax.device_count(), random_shuffle=True)
+            res = fn.resize(decode, resize_width=224, resize_height=224,
```

Comment:
For resize we are converting to NCHW and then again the layout is changed to NHWC after CMN, why is this transition required?